docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+# 10101 Architecture
+
+This is a working document to design the architecture drafted in the below diagram. It will not remain in that structure but rather will be eventually documented in ADRs.
+
+![Assumed component diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/get10101/10101/d7619613324df3ce5c0f0a35d97bf24323f9515b/docs/diagrams/testing-component-diagram.puml)
```

Comment:
Please consider changing this diagram to use mermaid's C4 diagrams - https://mermaid.js.org/syntax/c4c.html - as discussed here:
https://github.com/get10101/10101/pull/30/files#r1082034689